redbot/cogs/permissions/__init__.py

```diff
@@ -8,6 +8,7 @@
     # the permissions commands themselves have rules added.
     # Automatic listeners being added in add_cog happen in arbitrary
     # order, so we want to circumvent that.
-    bot.add_listener(cog.red_cog_added, "on_cog_add")
-    bot.add_listener(cog.red_command_added, "on_command_add")
+    await cog.on_cog_add(cog)
+    for command in cog.walk_commands():
```

Comment:
This needs to be swapped, `walk_commands` includes aliases, and exists for other purposes (mainly so that cogs mirror the bot method from discord.py)
```suggestion
    for command in cog.__cog_commands__:
```

Reply:
Ah I see. I chose `walk_commands()` because it's a public+documented interface, turns out it's not documented very well :D